x/subscription/keeper/subscription.go

```diff
@@ -381,14 +402,141 @@ func (k Keeper) resetSubscriptionDetailsAndAppendEntry(ctx sdk.Context, sub *typ
 
 	err := k.subsFS.AppendEntry(ctx, sub.Consumer, block, sub)
 	if err != nil {
+		// Remove new timer if failed
+		k.subsTS.DelTimerByBlockTime(ctx, expiry, []byte(sub.Consumer))
+
 		// normally would panic! but ok to ignore - the subscription remains
 		// as is with same remaining duration (but not renewed CU)
```

Comment:
I think it’s better to learn to use the transient keeper instead of reverting, we can set a snapshot of the keeper and dismiss it if we need just like transactions automatically do

Reply:
@omerlavanet That's a good idea!
We can do this in a new PR, going through all the `BeginBlock` functions, and implement it for them.

Reply:
opened a task. Resolving